File: javascript/find_outlier(6).js
function findOutlier(integers){
  var odd = [];
  var even = [];
  integers.map(function(num){
    if(num%2==1){
      odd.push(Math.abs(num));
    }
    else{
      even.push(Math.abs(num));
    };
  });
  if(odd.length==1){
    return odd[0];
  }
  else{
    return even[0];
  }
}

File: ruby/find_outlier(6).rb
def find_outlier(integers)
  even,odd = integers.partition { |num| num.even? }
  even.length > odd.length ? odd[0] : even[0]
end


print find_outlier([2,6,8,10,3])




# Test.assert_equals(find_outlier([0, 1, 2]), 1)
# Test.assert_equals(find_outlier([1, 2, 3]), 2)
# Test.assert_equals(find_outlier([2,6,8,10,3]), 3)
